### DISCUSSION TITLE
Make Edges easier to be selected

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#1268] [General] [answered]
Normally it is very hard to select an Edge in GLSP as you need to exactly click on the edge:

![Peek 2024-02-21 21-31](https://github.com/eclipse-glsp/glsp/assets/1339124/4971b788-9783-47c4-bbaf-46b968e24f25)

Now I found a solution to make it easier to select an edge. 
In my Edge View method `renderLine` I render two path elements embedded in one `<g>` element.

```javascript
 protected override renderLine(edge: GEdge, segments: Point[], context: RenderingContext, args?: IViewArgs): VNode {
        let path = '';
       ....
       ............
        const vnode: any = <g><path class-sprotty-edge={true} class-line={true} d={path} /><path class-edge-border={true} d={path} /></g>
        return vnode;
    }
```
The second path element is an invisible border with the following style making it transparent but with 15 px width : 

```css
.edge-border {
    stroke: transparent;
    stroke-width: 15px;
}
```

now you can select the edge much easier: 

![Peek 2024-02-21 21-33](https://github.com/eclipse-glsp/glsp/assets/1339124/1d13dcaa-a9e6-4637-8b0e-a111e53eeb3e)

(To be honest, the idea with a second transparent but wider path element came from ChatGPT)

What do you think about this solution? Has anyone else the problem that edges are hard to be selected?

**Comment by martin-fleck-at** [comment, marked as answer]
Hi Ralph,

Good news! The feature you mentioned has already been implemented as part of the [GEdge](https://github.com/eclipse-glsp/glsp-client/blob/e7dec9bd52b9688a7a23005c3f7de652d0e85923/packages/client/src/views/gedge-view.tsx#L52). We're grateful for your input — it's suggestions like yours that help us grow and improve. Could you please take a moment to double-check whether the existing implementation covers your use case as well? You can see how it is used in the [Java Server when the edge is created](https://github.com/eclipse-glsp/glsp-server/blob/2bb9e0b545f9f2657b26c8526f16a40ef5d2228e/examples/org.eclipse.glsp.example.workflow/src/org/eclipse/glsp/example/workflow/handler/CreateEdgeHandler.java#L40). Any insight or feedback you might have is greatly appreciated.

**Reply by rsoika** [reply]
Ok - I was not aware of this new feature. Of course your solution works much more precisely (by adding the class mouse-handle). 
I adapted the feature into our [own custom Renderer](https://github.com/imixs/open-bpmn/blob/master/open-bpmn.glsp-client/open-bpmn-glsp/src/bpmn-routing-views.tsx) now. The Result is brilliant! Thanks a lot.
